docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md

```diff
@@ -0,0 +1,678 @@
+2. In your Flutter app's `pubspec.yaml`, add the following dependencies:
+
+   Get the latest version from `https://pub.dev/packages/appium_flutter_server/install`
+
+   ```yaml
+   dev_dependencies:
+     appium_flutter_server: 0.0.14
+   ```
+
+3. Create a directory called `integration_test` in the root of your Flutter project.
+4. Create a file called `appium_test.dart` in the `integration_test` directory.
+5. Add the following code to the `appium_test.dart` file:
+
+   ```dart
+   import 'package:appium_flutter_server/appium_flutter_server.dart';
+   import 'package:appium_testing_app/main.dart';
+
+   void main() {
+     initializeTest(app: const MyApp());
+   }
+   ```
+   If you are in need to configure certain prerequists before the testing app is loaded, you can try the following code:
+   ```dart
+   import 'package:appium_testing_app/main.dart'; as app;
+   void main() {
+     initializeTest(
+       callback: (WidgetTester tester) async {
+          // Perform any prerequisite steps or intialise any dependencies required by the app
+          // and make sure to pump the app widget using below statement.
+          await tester.pumpWidget(const app.MyApp());
+       },
+     );
+   }
+   ```
+
+6. Build the Android app:
+
+   ```bash
+   ./gradlew app:assembleDebug -Ptarget=`pwd`/../integration_test/appium_test.dart
+   ```
+   
+   The APK file will be located in `{project-root}/build/app/outputs/flutter-apk/`
+
+7. Build the iOS app:
+   For Simulator - Debug mode
+   ```bash
+   flutter build ios integration_test/appium_test.dart --simulator
+   ```
+   
+   The app will be located in `{project-root}/build/ios/iphonesimulator/`
+
+   For Real Device - Release mode
+   ```bash
+    flutter build ipa --release integration_test/appium_test.dart
+    ```
+   
+   The IPA file will be located in `{project-root}/build/ios/ipa/`
+
+Bingo! You are ready to run your tests using Appium Flutter Integration Driver.
+
+## Uploading your Flutter App to Sauce Labs
+
+You can now upload the built apps with our [REST API](/dev/api/storage/#upload-file-to-app-storage), or manually upload them to the preferred Data Center. See [Manually Uploading an App](/mobile-apps/live-testing/live-mobile-app-testing/#uploading-an-app) for more information.
+
+## Configuring your Appium Capabilities
+
+More information on how to write Appium tests for Flutter apps can be found in the [appium-flutter-integration-driver](https://github.com/AppiumTestDistribution/appium-flutter-integration-driver) repository. You can also find a sample Flutter app and tests in the [Demo JS - Appium Flutter Integration](https://github.com/saucelabs-training/demo-js/tree/main/webdriverio/appium-app/examples/appium-flutter-integration)-repository.
+
+:::info W3C Capabilities
+We encourage you to use W3C capabilities for your tests. For more information on W3C capabilities, see [W3C Capabilities](/dev/test-configuration-options/#mobile-appium-capabilities).
+:::
```

Comment:
Is this needed? Appium 2 only supports W3C anyway.

Reply:
true! will remove it now